treatment-service/models/storage.go

```diff
@@ -384,15 +397,22 @@ func NewExperimentIndex(experiment *pubsub.Experiment) *ExperimentIndex {
 					realSets[key] = set.New()
 				}
 				realSets[key].Insert(val.GetReal())
+			case *_segmenters.SegmenterValue_Bool:
+				boolFlags[key] = val.GetBool()
 			}
 		}
 	}
 
+	// Delete all segments since they have already been converted to the various sets stored in ExperimentIndex,
+	// and are no longer used by the Treatment Service
+	experiment.Segments = nil
```

Comment:
Could you leave a `TODO` note here with the long term idea of letting the `ExperimentIndex` store only the relevant data using appropriate structs rather than attempting to reuse this pubsub message type and deleting the redundant data from it?

Reply:
Added the `TODO` note just below that comment!